src/components/pages/import-batch-view.js

```diff
@@ -25,34 +25,41 @@ function ImportBatchView({ initialState }) {
 
   const [batch, setBatch] = useState(initialState.batch);
 
-  const id = batch._id;
-  const isCompletedImport = batch.progress === 1;
+  const { _id: id, progress } = batch;
   const { hostName, allowUnsecure } = batch.batchParams;
 
   useEffect(() => {
-    if (isCompletedImport) {
-      return;
-    }
+    let nextTimeout = null;
+
+    const getUpdate = async () => {
+      if (progress === 1) {
+        if (nextTimeout) {
+          clearTimeout(nextTimeout);
+        }
+        return;
+      }
 
-    const interval = setInterval(async () => {
       try {
         const updatedBatchDetails = await importApiClient.getImportBatch(id);
         setBatch(updatedBatchDetails.batch);
-        if (updatedBatchDetails.batch.progress === 1) {
-          clearInterval(interval);
-        }
+        nextTimeout = setTimeout(getUpdate, POLL_INTERVAL_IN_MS);
       } catch (error) {
         handleApiError({ error, logger, t });
+        if (nextTimeout) {
+          clearTimeout(nextTimeout);
+        }
       }
-    }, POLL_INTERVAL_IN_MS);
+    };
 
-    // eslint-disable-next-line consistent-return
+    if (!nextTimeout) {
```

Comment:
This condition can never be false as `nextTimeout` is set to `null` two statements above (line 32) with no async code and therefore no thread change in between.